packages/features/ee/round-robin/roundRobinReassignment.ts

```diff
@@ -0,0 +1,517 @@
+      };
+    } else if (cancelledRRHostEvt.team) {
+      // Filter out the new RR host from attendees and add the old RR host
+      const newMembersArray = cancelledRRHostEvt.team?.members || [];
+      cancelledRRHostEvt.team.members = newMembersArray.filter(
+        (member) => member.email !== reassignedRRHost.email
+      );
+      cancelledRRHostEvt.team.members.unshift({
+        id: previousRRHost.id,
+        email: previousRRHost.email,
+        name: previousRRHost.name || "",
+        timeZone: previousRRHost.timeZone,
+        language: { translate: previousRRHostT, locale: previousRRHost.locale || "en" },
+      });
+    }
+
+    await sendRoundRobinCancelledEmails(cancelledRRHostEvt, [
+      {
+        ...previousRRHost,
+        name: previousRRHost.name || "",
+        username: previousRRHost.username || "",
+        timeFormat: getTimeFormatStringFromUserTimeFormat(previousRRHost.timeFormat),
+        language: { translate: previousRRHostT, locale: previousRRHost.locale || "en" },
+      },
+    ]);
+  }
+
+  // Handle changing workflows with organizer
+  if (hasOrganizerChanged) {
+    const workflowReminders = await prisma.workflowReminder.findMany({
+      where: {
+        bookingUid: booking.uid,
+        method: WorkflowMethods.EMAIL,
+        workflowStep: {
+          action: WorkflowActions.EMAIL_HOST,
+          workflow: {
+            trigger: {
+              in: [
+                WorkflowTriggerEvents.BEFORE_EVENT,
+                WorkflowTriggerEvents.NEW_EVENT,
+                WorkflowTriggerEvents.AFTER_EVENT,
+              ],
+            },
+          },
+        },
+      },
+      include: {
+        workflowStep: {
+          select: {
+            template: true,
+            workflow: {
+              select: {
+                trigger: true,
+                time: true,
+                timeUnit: true,
+              },
+            },
+          },
+        },
+      },
+    });
+
+    for (const workflowReminder of workflowReminders) {
+      const workflowStep = workflowReminder?.workflowStep;
+      const workflow = workflowStep?.workflow;
+
+      if (workflowStep && workflow) {
+        await scheduleEmailReminder({
+          evt: {
+            ...evt,
+            eventType,
+          },
+          action: WorkflowActions.EMAIL_HOST,
+          triggerEvent: workflow.trigger,
+          timeSpan: {
+            time: workflow.time,
+            timeUnit: workflow.timeUnit,
+          },
+          sendTo: reassignedRRHost.email,
+          template: workflowStep.template,
+        });
+      }
+
+      await deleteScheduledEmailReminder(workflowReminder.id, workflowReminder.referenceId);
+    }
+    // Send new event workflows to new organizer
+    const newEventWorkflows = await prisma.workflow.findMany({
+      where: {
+        trigger: WorkflowTriggerEvents.NEW_EVENT,
+        OR: [
+          {
+            isActiveOnAll: true,
```

Comment:
I think we need to add something like this: 

- For team workflows: 

```
OR: {
	isActiveOnAll: true,
	teamId: eventType.teamId
}
```

- For org workflows: 

```
OR: {
	isActiveOnAll: true,
	teamId: eventType.team.parent.id
}
```